File: 02-february/09-longest-balanced-subarray-i/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestBalanced = function(nums) {
  const n = nums.length;
  let best = 0;

  for (let left = 0; left < n; left++) {
    const evenSeen = new Set();
    const oddSeen = new Set();

    for (let right = left; right < n; right++) {
      const value = nums[right];
      if (value % 2 === 0) {
        evenSeen.add(value);
      } else {
        oddSeen.add(value);
      }

      if (evenSeen.size === oddSeen.size) {
        best = Math.max(best, right - left + 1);
      }
    }
  }

  return best;
};

File: 02-february/09-longest-balanced-subarray-i/solution.py
class Solution:
    def longestBalanced(self, nums: List[int]) -> int:
        n = len(nums)
        best = 0

        for left in range(n):
            even_seen = set()
            odd_seen = set()

            for right in range(left, n):
                value = nums[right]
                if value % 2 == 0:
                    even_seen.add(value)
                else:
                    odd_seen.add(value)

                if len(even_seen) == len(odd_seen):
                    best = max(best, right - left + 1)

        return best
